client/landing/gutenboarding/onboarding-block/edit.tsx

```diff
@@ -35,12 +35,25 @@ const OnboardingEdit: React.FunctionComponent< BlockEditProps< Attributes > > =
 
 	const makePath = usePath();
 
-	const { pathname } = useLocation();
+	const { pathname, state: locationState = {} } = useLocation< GutenLocationStateType >();
 
 	React.useEffect( () => {
 		setTimeout( () => window.scrollTo( 0, 0 ), 0 );
 	}, [ pathname ] );
 
+	// makePathState( path: StepType ) - A wrapper around makePath() that preserves location state.
+	// This uses makePath() to generate a string path, then transforms that
+	// string path into an object that also contains the location state.
+	const makePathState = React.useCallback(
```

Comment:
Picky, but for readability could this be `makePathWithState`?

Reply:
Sure, done in b877b4b976